### DISCUSSION TITLE
Using third party library with Kotlin + Unreal

### DISCUSSION THREAD

**Discussion by Eecornwell** [hiperbou/kotlin-unreal#27] [Q&A] [answered]
Hello,
I have a third party library (.aar) that I have imported into Unreal Engine v4.27 using UPL for Android.  Am I able to call the third party library using Kotlin and your solution?  Is this something you have tried or enabled out of the box?  I was able to get the .aar + Kotlin working in Android Studio and am looking to make the leap to Unreal import.

**Comment by hiperbou** [comment, marked as answer]
Hello,
Although this project uses the Kotlin language, it doesn't have anything to do with native Android development. So it will not make you able to call third party android libraries.

My guess is that you should look for JNI and Java/Kotlin interoperability with Unreal C++. A quick search on google lead me to this:
https://github.com/Sovahero/PluginMobileNativeCode

This could be useful too:
https://github.com/VRMonkey/NativeAndroidUE4Plugin
https://isaratech.com/ue4-making-a-android-plugin-in-10-minutes/

Good luck!
